util/dNNInterfaces/MLfoyInterface.py

```diff
@@ -456,49 +481,70 @@ def evtYieldCategories():
         # writing code for dnn output plots
         string += """
 def plots_dnn(data, discrname):
-    categories = []
-    nhistobins = []
-    minxvals = []
-    maxxvals = []
-    discrs = []
 
-    ndefaultbins = {}\n\n""".format(ndefaultbins)
+    ndefaultbins = {}
+    category_dict = {{}}
+    this_dict = {{}}\n\n""".format(ndefaultbins)
+
 
         # loop over dnns witing code for DNN discr plots
         for dnn in self.DNNs:
-            string += dnn.generateOutputPlots()
+            string += dnn.generateOutputPlots(variable_binning = variable_binning, nbins = ndefaultbins)
 
         # generating plot classes for DNN plots and adding info to data class
         string += """
-    plotPreselections = [c[0] for c in categories]
-    binlabels =         [c[1] for c in categories]
-
-    DNNPlots = []
-    for discr, sel, label, nbins, minx, maxx in zip(discrs,plotPreselections,binlabels,nhistobins,minxvals,maxxvals):
-        DNNPlots.append(
-            plotClasses.Plot(
-                ROOT.TH1F(discrname+"_"+label,"final discriminator ("+label+")",nbins,minx,maxx),
-                discr,sel,label))
-
-    data.categories += categories
-    data.discrs     += discrs
-    data.nhistobins += nhistobins
-    data.minxvals   += minxvals
-    data.maxxvals   += maxxvals
-    
-    data.plotPreselections  += plotPreselections
-    data.binlabels          += binlabels
 
+    for l in this_dict:
+        this_dict[l]["histoname"] = this_dict[l]["discr"]+"_"+l
```

Comment:
So Philip, I am very sad to say but I found an error in your merge request after all. It really saddens me to see this, but don't worry, I will fix this with a simple bugfix-push.
The histograms you are naming here with `this_dict[l]["discr"]+"_"+l`will in the end look like this `DNNOutput_JTCATEGORY_NODENAME_ljets_JTCATEGORY_NODENAME_XXX` which is not at all what we want.
The right naming scheme should be `finaldiscr_ljets_JTCATEGORY_NODENAME_XXX`, which you can get with the `discrname` variable you give as an argument to this function.
So the right way to initialize your histogram names would be `discrname+"_"+l`

Cheers,
Jan